app/models/name.rb

```diff
@@ -418,16 +431,26 @@ def best_brief_description
   # Used by show_name.
   def self.count_observations(names)
     ids = names.map(&:id)
-    counts_and_ids = Name.connection.select_rows(%(
-        SELECT count(*) c, names.id i FROM observations, names
-        WHERE observations.name_id = names.id
-        AND names.id IN (#{ids.join(", ")}) group by names.id
-    ))
+    select_manager = arel_select_count_observations
+
+    counts_and_ids = Name.connection.select_rows(select_manager.to_sql)
     result = {}
     counts_and_ids.each { |row| result[row[1]] = row[0] }
     result
   end
```

Comment:
Ooh, this can be translated into a single line!

  Hash[*Observation.group(:name_id).where(name: names).pluck(:name_id, "count(*)").each]